artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java

```diff
@@ -196,6 +196,10 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
          }
 
          ProtocolManager protocolManagerToUse = protocolMap.get(protocolToUse);
+         if (protocolManagerToUse == null) {
+            ActiveMQServerLogger.LOGGER.failedToFindProtocolManager(ctx.channel().remoteAddress().toString(), ctx.channel().localAddress().toString(), protocolToUse, protocolMap.keySet().toString());
```

Comment:
Question, how sure/guarenteed is it that these objects will def be wont be null them selves, aka doing this logger wont cause an NPE.

ctx.channel().remoteAddress().toString()
ctx.channel().localAddress().toString()

is it worth null checking the dot train?

e.g. is it worth doing?

ctx.channel() == null ? null : ctx.channel().localAddress() == null ? null : ctx.channel().localAddress().toString

Reply:
I performed manual testing to confirm the logging works as expected, but I'm not 100% sure these won't *ever* be null so checks are worth doing.